compiler/noirc_evaluator/src/ssa/ir/dfg/simplify/call.rs

```diff
@@ -91,14 +91,18 @@ pub(super) fn simplify_call(
             }
         }
         Intrinsic::ArrayLen => {
-            if let Some(length) = dfg.try_get_array_length(arguments[0]) {
-                let length = FieldElement::from(u128::from(length));
-                SimplifyResult::SimplifiedTo(dfg.make_constant(length, NumericType::length_type()))
-            } else if matches!(dfg.type_of_value(arguments[1]), Type::Slice(_)) {
-                SimplifyResult::SimplifiedTo(arguments[0])
-            } else {
-                SimplifyResult::None
-            }
+            let length = match dfg.type_of_value(arguments[0]) {
+                Type::Array(_, length) => {
+                    let length = FieldElement::from(u128::from(length));
+                    dfg.make_constant(length, NumericType::length_type())
+                }
```

Comment:
Maybe we just add a unit test here that we simplify into what we expect and then we can avoid these mutation warnings

Reply:
Good idea, done!